### PR TITLE
feat(ingest/cli): add undo soft delete command

### PR DESCRIPTION
**Pull request by anshbansal**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/graph/client.py [hunk 1]
```diff
@@ -1241,14 +1241,29 @@ def soft_delete_entity(
         Args:
             urn: The urn of the entity to soft-delete.
         """
+        self.change_soft_delete_status(
+            urn=urn, run_id=run_id, deletion_timestamp=deletion_timestamp, delete=True
+        )
+
+    def change_soft_delete_status(
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/graph/client.py:1248]:
```suggestion
    def set_soft_delete_status(
```

File: metadata-ingestion/src/datahub/ingestion/graph/client.py [hunk 2]
```diff
@@ -1241,14 +1241,29 @@ def soft_delete_entity(
...
+        self,
+        urn: str,
+        delete: bool,
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/graph/client.py:1251]:
```suggestion
        deleted: bool,
```

File: metadata-ingestion/src/datahub/cli/delete_cli.py [hunk 3]
```diff
@@ -213,6 +213,17 @@ def references(urn: str, dry_run: bool, force: bool) -> None:
         logger.info(f"Deleted {references_count} references to {urn}")
 
 
+@delete.command()
+@click.option("--urn", required=True, type=str, help="the urn of the entity")
+def undo_by_filter(urn: str) -> None:
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/cli/delete_cli.py:218]: there's no filters supported here - so the `undo_by_filter` method name is a bit strange

File: metadata-ingestion/src/datahub/cli/delete_cli.py [hunk 4]
```diff
@@ -213,6 +213,17 @@ def references(urn: str, dry_run: bool, force: bool) -> None:
...
+    """
+    Allows to un-soft delete by filters
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/cli/delete_cli.py:220]:
```suggestion
    Undo a soft deletion of an entity.
```

**Review by hsheth2 — APPROVED** [review]

File: metadata-ingestion/src/datahub/cli/delete_cli.py (hunk 3 above)
Comment (anshbansal) [reply metadata-ingestion/src/datahub/cli/delete_cli.py:218]: it allows us to add them in future

File: metadata-ingestion/src/datahub/ingestion/graph/client.py (hunk 2 above)
Comment (anshbansal) [reply metadata-ingestion/src/datahub/ingestion/graph/client.py:1251]: delete is correct here

File: metadata-ingestion/src/datahub/ingestion/graph/client.py (hunk 1 above)
Comment (anshbansal) [reply metadata-ingestion/src/datahub/ingestion/graph/client.py:1248]: will incorporate

File: metadata-ingestion/src/datahub/cli/delete_cli.py (hunk 4 above)
Comment (anshbansal) [reply metadata-ingestion/src/datahub/cli/delete_cli.py:220]: will incorporate

**Pull request merged**